src/overlays/actors/ovl_En_GirlA/z_en_girla.c

```diff
@@ -20,139 +74,901 @@ const ActorInit En_GirlA_InitVars = {
+    { 0x00D8, 0x0024, func_8002EBCC, 0x0014, 0x000A, 0x00A0, 0x008A, GI_ARROWS_SMALL, func_80A3AAA8, func_80A3B634, func_80A3BB6C },
+    { 0x00CE, 0x001F, func_8002EBCC, 0x0050, 0x0014, 0x001C, 0x0006, GI_BOMBS_20, func_80A3AB58, func_80A3B678, func_80A3BB6C },
+    { 0x00CE, 0x001F, func_8002EBCC, 0x0078, 0x001E, 0x001D, 0x001E, GI_BOMBS_30, func_80A3AB58, func_80A3B678, func_80A3BB6C },
+    { 0x00CE, 0x001F, func_8002EBCC, 0x0023, 0x0005, 0x00CB, 0x00CA, GI_BOMBS_5, func_80A3AB58, func_80A3B678, func_80A3BB6C },
+    { 0x00EB, 0x0037, func_8002EBCC, 0x0028, 0x0001, 0x0064, 0x0062, GI_POTION_RED, func_80A3ADD4, func_80A3BA40, func_80A3BB6C },
+    { 0x00EB, 0x0037, func_8002EBCC, 0x0032, 0x0001, 0x0065, 0x0063, GI_POTION_RED, func_80A3ADD4, func_80A3BA40, func_80A3BB6C },
+};
 
+s16 D_80A3CCB4[8] = { 0x0005, 0x000A, 0x000F, 0x0014, 0x0019, 0x001E, 0x0023, 0x0028 };
+
+void func_80A3A750(EnGirlA* this, void* arg1) {
+    this->actionFunc = arg1;
+}
+
+s32 func_80A3A758(EnGirlA* this) {
+    switch (this->actor.params) {
+        case 0x11:
+            if (gSaveContext.itemGetInf[0] & 0x4) {
+                this->actor.params = 0x10;
+                return 1;
+            }
+            break;
+        case 0x18:
+            if (gSaveContext.itemGetInf[0] & 0x40) {
+                this->actor.params = 0x26;
+                return 1;
+            }
+            break;
+        case 0x19:
+            if (gSaveContext.itemGetInf[0] & 0x80) {
+                this->actor.params = 0x26;
+                return 1;
+            }
+            break;
+        case 0x1A:
+            if (gSaveContext.itemGetInf[0] & 0x100) {
+                this->actor.params = 0x26;
+                return 1;
+            }
+            break;
+        case 0x1B:
+            if (gSaveContext.itemGetInf[0] & 0x200) {
+                this->actor.params = 0x26;
+                return 1;
+            }
+            break;
+        case 0x1C:
+            if (gSaveContext.itemGetInf[0] & 0x400) {
+                this->actor.params = 0x26;
+                return 1;
+            }
+            break;
+        case 0x15:
+            if (gSaveContext.itemGetInf[0] & 0x8) {
+                this->actor.params = 0x26;
+                return 1;
+            }
+            break;
+        case 0x16:
+            if (gSaveContext.itemGetInf[0] & 0x10) {
+                this->actor.params = 0x26;
+                return 1;
+            }
+            break;
+        case 0x17:
+            if (gSaveContext.itemGetInf[0] & 0x20) {
+                this->actor.params = 0x26;
+                return 1;
+            }
+            break;
+    }
+    return 0;
+}
+
+#ifdef NON_MATCHING
+// Weird allocations
+void func_80A3A8D0(EnGirlA* this, GlobalContext* globalCtx) {
+    char** strptr = &D_80A3C590[this->actor.params];
+    s16 params = this->actor.params;
+
+    if (strptr) {};
+    osSyncPrintf("%s(%2d)初期設定\n", *(&D_80A3C590[this->actor.params]), this->actor.params);
```

Comment:
```suggestion
    osSyncPrintf("%s(%2d)初期設定\n", D_80A3C590[this->actor.params], this->actor.params);
```

Reply:
I'm leaving this as-is, because that's the closest I can get to matching that function.